rust/worker/src/segment/record_segment.rs

```diff
@@ -937,3 +937,111 @@ impl RecordSegmentReader<'_> {
             .await
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::sync::{atomic::AtomicU32, Arc};
+
+    use chroma_blockstore::BlockfileWriter;
+    use chroma_types::Chunk;
+    use shuttle::{future, thread};
+
+    use crate::{
+        log::test::{upsert_generator, LogGenerator},
+        segment::{
+            record_segment::MAX_OFFSET_ID, test::TestSegment, LogMaterializer, SegmentWriter,
+        },
+    };
+
+    use super::RecordSegmentWriter;
+
+    // The same record segment writer should be able to run concurrently on different threads without conflict
+    #[test]
+    fn test_max_offset_id_shuttle() {
+        shuttle::check_random(
+            || {
+                let log_partition_size = 1000;
+                let stack_size = 1 << 22;
+                let thread_count = num_cpus::get() * 2;
```

Comment:
now that i am thinking about it, maybe this should be logged or set to a fixed value. Otherwise if a bug is discovered in CI a shuttle repro will not be 1:1 with ones local machine.